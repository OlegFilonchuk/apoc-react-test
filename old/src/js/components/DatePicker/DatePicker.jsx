import React from 'react';
import PropTypes from 'prop-types';
import DateTime from 'react-datetime';
import moment from 'moment';

import 'react-datetime/css/react-datetime.css';

class DatePicker extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: this.props.value
    };
    this.handleChange = this.handleChange.bind(this);
  }

  componentWillReceiveProps() {
    this.setState((previousState, props) => {
      const value = moment(props.value);

      return { value };
    });
  }

  handleChange(date) {
    const { onChange } = this.props;

    this.setState(
      () => ({ value: date }),
      () => {
        onChange(date.toDate());
      }
    );
  }

  render() {
    const { onChange, dateFormat, timeFormat, closeOnSelect, ...props } = this.props;

    return (
      <DateTime
        {...props}
        dateFormat={dateFormat}
        timeFormat={timeFormat}
        closeOnSelect={closeOnSelect}
        onChange={this.handleChange}
        value={this.state.value}
      />
    );
  }
}

DatePicker.propTypes = {
  onChange: PropTypes.func,
  /* eslint-disable react/forbid-prop-types */
  value: PropTypes.object,
  /* eslint-enable react/forbid-prop-types */
  dateFormat: PropTypes.string,
  timeFormat: PropTypes.bool,
  closeOnSelect: PropTypes.bool
};

DatePicker.defaultProps = {
  onChange: () => true,
  dateFormat: 'MM/DD/YYYY',
  value: moment(),
  timeFormat: false,
  closeOnSelect: true
};

export default DatePicker;
