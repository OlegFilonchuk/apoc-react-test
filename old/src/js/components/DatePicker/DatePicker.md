This component wraps react-datetime [DateTime](https://github.com/arqex/react-datetime)
Additional props that can be passed to the component can be found in the
DateTime [props definition](https://github.com/arqex/react-datetime#api)

## Simple DatePicker

    (
      <DatePicker />
    )

