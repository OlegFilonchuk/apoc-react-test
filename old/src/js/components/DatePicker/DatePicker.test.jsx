import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import moment from 'moment';
import DatePicker from './DatePicker';

describe('<DatePicker>', () => {
  const mountDatePicker = (props, mountFn = mount) => mountFn(<DatePicker {...props} />);

  it('should render DatePicker component', () => {
    const props = {
      value: moment.utc('2017-04-29')
    };
    const component = mountDatePicker(props);

    expect(toJson(component)).toMatchSnapshot();
  });
});
