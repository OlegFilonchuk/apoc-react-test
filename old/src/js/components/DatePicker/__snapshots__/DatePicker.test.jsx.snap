// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker> should render DatePicker component 1`] = `
<DatePicker
  closeOnSelect={true}
  dateFormat="MM/DD/YYYY"
  onChange={[Function]}
  timeFormat={false}
  value={"2017-04-29T00:00:00.000Z"}
>
  <Component
    className=""
    closeOnSelect={true}
    closeOnTab={true}
    dateFormat="MM/DD/YYYY"
    defaultValue=""
    input={true}
    inputProps={Object {}}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onViewModeChange={[Function]}
    strictParsing={true}
    timeConstraints={Object {}}
    timeFormat={false}
    utc={false}
    value={"2017-04-29T00:00:00.000Z"}
  >
    <div
      className="rdt"
    >
      <input
        className="form-control"
        key="i"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="04/29/2017"
      />
      <div
        className="rdtPicker"
        key="dt"
      >
        <Component
          onClickOutside={[Function]}
          view="days"
          viewProps={
            Object {
              "addTime": [Function],
              "dateFormat": "MM/DD/YYYY",
              "handleClickOutside": [Function],
              "isValidDate": undefined,
              "localMoment": [Function],
              "renderDay": undefined,
              "renderMonth": undefined,
              "renderYear": undefined,
              "selectedDate": "2017-04-29T00:00:00.000Z",
              "setDate": [Function],
              "setTime": [Function],
              "showView": [Function],
              "subtractTime": [Function],
              "timeConstraints": Object {},
              "timeFormat": "",
              "updateOn": "days",
              "updateSelectedDate": [Function],
              "value": "2017-04-29T00:00:00.000Z",
              "viewDate": "2017-04-01T00:00:00.000Z",
            }
          }
        >
          <OnClickOutside(Component)
            addTime={[Function]}
            dateFormat="MM/DD/YYYY"
            eventTypes={
              Array [
                "mousedown",
                "touchstart",
              ]
            }
            excludeScrollbar={false}
            handleClickOutside={[Function]}
            isValidDate={undefined}
            localMoment={[Function]}
            outsideClickIgnoreClass="ignore-react-onclickoutside"
            preventDefault={false}
            renderDay={undefined}
            renderMonth={undefined}
            renderYear={undefined}
            selectedDate={"2017-04-29T00:00:00.000Z"}
            setDate={[Function]}
            setTime={[Function]}
            showView={[Function]}
            stopPropagation={false}
            subtractTime={[Function]}
            timeConstraints={Object {}}
            timeFormat=""
            updateOn="days"
            updateSelectedDate={[Function]}
            value={"2017-04-29T00:00:00.000Z"}
            viewDate={"2017-04-01T00:00:00.000Z"}
          >
            <Component
              addTime={[Function]}
              dateFormat="MM/DD/YYYY"
              disableOnClickOutside={[Function]}
              enableOnClickOutside={[Function]}
              eventTypes={
                Array [
                  "mousedown",
                  "touchstart",
                ]
              }
              handleClickOutside={[Function]}
              isValidDate={undefined}
              localMoment={[Function]}
              outsideClickIgnoreClass="ignore-react-onclickoutside"
              preventDefault={false}
              renderDay={undefined}
              renderMonth={undefined}
              renderYear={undefined}
              selectedDate={"2017-04-29T00:00:00.000Z"}
              setDate={[Function]}
              setTime={[Function]}
              showView={[Function]}
              stopPropagation={false}
              subtractTime={[Function]}
              timeConstraints={Object {}}
              timeFormat=""
              updateOn="days"
              updateSelectedDate={[Function]}
              value={"2017-04-29T00:00:00.000Z"}
              viewDate={"2017-04-01T00:00:00.000Z"}
            >
              <div
                className="rdtDays"
              >
                <table>
                  <thead
                    key="th"
                  >
                    <tr
                      key="h"
                    >
                      <th
                        className="rdtPrev"
                        key="p"
                        onClick={[Function]}
                      >
                        <span>
                          ‹
                        </span>
                      </th>
                      <th
                        className="rdtSwitch"
                        colSpan={5}
                        data-value={3}
                        key="s"
                        onClick={[Function]}
                      >
                        April 2017
                      </th>
                      <th
                        className="rdtNext"
                        key="n"
                        onClick={[Function]}
                      >
                        <span>
                          ›
                        </span>
                      </th>
                    </tr>
                    <tr
                      key="d"
                    >
                      <th
                        className="dow"
                        key="Su0"
                      >
                        Su
                      </th>
                      <th
                        className="dow"
                        key="Mo1"
                      >
                        Mo
                      </th>
                      <th
                        className="dow"
                        key="Tu2"
                      >
                        Tu
                      </th>
                      <th
                        className="dow"
                        key="We3"
                      >
                        We
                      </th>
                      <th
                        className="dow"
                        key="Th4"
                      >
                        Th
                      </th>
                      <th
                        className="dow"
                        key="Fr5"
                      >
                        Fr
                      </th>
                      <th
                        className="dow"
                        key="Sa6"
                      >
                        Sa
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    key="tb"
                  >
                    <tr
                      key="4_1"
                    >
                      <td
                        className="rdtDay rdtOld"
                        data-value={26}
                        key="3_26"
                        onClick={[Function]}
                      >
                        26
                      </td>
                      <td
                        className="rdtDay rdtOld"
                        data-value={27}
                        key="3_27"
                        onClick={[Function]}
                      >
                        27
                      </td>
                      <td
                        className="rdtDay rdtOld"
                        data-value={28}
                        key="3_28"
                        onClick={[Function]}
                      >
                        28
                      </td>
                      <td
                        className="rdtDay rdtOld"
                        data-value={29}
                        key="3_29"
                        onClick={[Function]}
                      >
                        29
                      </td>
                      <td
                        className="rdtDay rdtOld"
                        data-value={30}
                        key="3_30"
                        onClick={[Function]}
                      >
                        30
                      </td>
                      <td
                        className="rdtDay rdtOld"
                        data-value={31}
                        key="3_31"
                        onClick={[Function]}
                      >
                        31
                      </td>
                      <td
                        className="rdtDay"
                        data-value={1}
                        key="4_1"
                        onClick={[Function]}
                      >
                        1
                      </td>
                    </tr>
                    <tr
                      key="4_8"
                    >
                      <td
                        className="rdtDay"
                        data-value={2}
                        key="4_2"
                        onClick={[Function]}
                      >
                        2
                      </td>
                      <td
                        className="rdtDay"
                        data-value={3}
                        key="4_3"
                        onClick={[Function]}
                      >
                        3
                      </td>
                      <td
                        className="rdtDay"
                        data-value={4}
                        key="4_4"
                        onClick={[Function]}
                      >
                        4
                      </td>
                      <td
                        className="rdtDay"
                        data-value={5}
                        key="4_5"
                        onClick={[Function]}
                      >
                        5
                      </td>
                      <td
                        className="rdtDay"
                        data-value={6}
                        key="4_6"
                        onClick={[Function]}
                      >
                        6
                      </td>
                      <td
                        className="rdtDay"
                        data-value={7}
                        key="4_7"
                        onClick={[Function]}
                      >
                        7
                      </td>
                      <td
                        className="rdtDay"
                        data-value={8}
                        key="4_8"
                        onClick={[Function]}
                      >
                        8
                      </td>
                    </tr>
                    <tr
                      key="4_15"
                    >
                      <td
                        className="rdtDay"
                        data-value={9}
                        key="4_9"
                        onClick={[Function]}
                      >
                        9
                      </td>
                      <td
                        className="rdtDay"
                        data-value={10}
                        key="4_10"
                        onClick={[Function]}
                      >
                        10
                      </td>
                      <td
                        className="rdtDay"
                        data-value={11}
                        key="4_11"
                        onClick={[Function]}
                      >
                        11
                      </td>
                      <td
                        className="rdtDay"
                        data-value={12}
                        key="4_12"
                        onClick={[Function]}
                      >
                        12
                      </td>
                      <td
                        className="rdtDay"
                        data-value={13}
                        key="4_13"
                        onClick={[Function]}
                      >
                        13
                      </td>
                      <td
                        className="rdtDay"
                        data-value={14}
                        key="4_14"
                        onClick={[Function]}
                      >
                        14
                      </td>
                      <td
                        className="rdtDay"
                        data-value={15}
                        key="4_15"
                        onClick={[Function]}
                      >
                        15
                      </td>
                    </tr>
                    <tr
                      key="4_22"
                    >
                      <td
                        className="rdtDay"
                        data-value={16}
                        key="4_16"
                        onClick={[Function]}
                      >
                        16
                      </td>
                      <td
                        className="rdtDay"
                        data-value={17}
                        key="4_17"
                        onClick={[Function]}
                      >
                        17
                      </td>
                      <td
                        className="rdtDay"
                        data-value={18}
                        key="4_18"
                        onClick={[Function]}
                      >
                        18
                      </td>
                      <td
                        className="rdtDay"
                        data-value={19}
                        key="4_19"
                        onClick={[Function]}
                      >
                        19
                      </td>
                      <td
                        className="rdtDay"
                        data-value={20}
                        key="4_20"
                        onClick={[Function]}
                      >
                        20
                      </td>
                      <td
                        className="rdtDay"
                        data-value={21}
                        key="4_21"
                        onClick={[Function]}
                      >
                        21
                      </td>
                      <td
                        className="rdtDay"
                        data-value={22}
                        key="4_22"
                        onClick={[Function]}
                      >
                        22
                      </td>
                    </tr>
                    <tr
                      key="4_29"
                    >
                      <td
                        className="rdtDay"
                        data-value={23}
                        key="4_23"
                        onClick={[Function]}
                      >
                        23
                      </td>
                      <td
                        className="rdtDay"
                        data-value={24}
                        key="4_24"
                        onClick={[Function]}
                      >
                        24
                      </td>
                      <td
                        className="rdtDay"
                        data-value={25}
                        key="4_25"
                        onClick={[Function]}
                      >
                        25
                      </td>
                      <td
                        className="rdtDay"
                        data-value={26}
                        key="4_26"
                        onClick={[Function]}
                      >
                        26
                      </td>
                      <td
                        className="rdtDay"
                        data-value={27}
                        key="4_27"
                        onClick={[Function]}
                      >
                        27
                      </td>
                      <td
                        className="rdtDay"
                        data-value={28}
                        key="4_28"
                        onClick={[Function]}
                      >
                        28
                      </td>
                      <td
                        className="rdtDay rdtActive"
                        data-value={29}
                        key="4_29"
                        onClick={[Function]}
                      >
                        29
                      </td>
                    </tr>
                    <tr
                      key="5_6"
                    >
                      <td
                        className="rdtDay"
                        data-value={30}
                        key="4_30"
                        onClick={[Function]}
                      >
                        30
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={1}
                        key="5_1"
                        onClick={[Function]}
                      >
                        1
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={2}
                        key="5_2"
                        onClick={[Function]}
                      >
                        2
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={3}
                        key="5_3"
                        onClick={[Function]}
                      >
                        3
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={4}
                        key="5_4"
                        onClick={[Function]}
                      >
                        4
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={5}
                        key="5_5"
                        onClick={[Function]}
                      >
                        5
                      </td>
                      <td
                        className="rdtDay rdtNew"
                        data-value={6}
                        key="5_6"
                        onClick={[Function]}
                      >
                        6
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </Component>
          </OnClickOutside(Component)>
        </Component>
      </div>
    </div>
  </Component>
</DatePicker>
`;
